src/unix/tcp.c

```diff
@@ -243,6 +248,29 @@ static int uv__ipv6_link_local_scope_id(void) {
   }
 
   freeifaddrs(ifa);
+#else
+  uv_interface_address_t* interfaces;
+  int count, i;
+
+  if (uv_interface_addresses(&interfaces, &count)) {
+    return 0;
+  }
+
+  rv = 0;
+
+  for (i = 0; i < count; i++) {
+    if (uv__is_ipv6_link_local((const struct sockaddr*) &interfaces[i].address.address6)) {
```

Comment:
You'll need to guard this on whether `interfaces[i].address->sa_family == AF_INET6` to avoid misinterpreting the address.

Reply:
hmm although now I see `uv__is_ipv6_link_local` already checks.